clang/test/CodeGen/nvptx-surface.cu

```diff
@@ -0,0 +1,3329 @@
+// RUN: %clang_cc1 -triple nvptx-unknown-unknown -fcuda-is-device -O3 -o - %s -emit-llvm | FileCheck %s
+// RUN: %clang_cc1 -triple nvptx64-unknown-unknown -fcuda-is-device -O3 -o - %s -emit-llvm | FileCheck %s
+#include "../Headers/Inputs/include/cuda.h"
```

Comment:
Should "include/cuda.h" be in `clang/test/CodeGen/Inputs`  because it's needed for a CodeGen test?

test/Headers contains (if I'm not mistaken) test cases related to headers, not header files for other tests. And it seems unusual for a test to reach into the input files of a different category of tests.

Reply:
Any suggestions for how to best not duplicate it?  I could copy it over, but that feels worse.

Reply:
Relative include paths are a problem for tests -- some run remotely with only the test directory staged, so there may not always be ../Headers directory where the tests actually run. Replicating test input changes is OK.

Reply:
https://github.com/llvm/llvm-project/pull/134459 has the fixup